src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -815,6 +715,113 @@ static time_t tryGetPartCreateTime(zkutil::ZooKeeperPtr & zookeeper, const Strin
 }
 
 
+void StorageReplicatedMergeTree::dropReplica(const String & replica, bool is_drop_table)
+{
+    auto zookeeper = tryGetZooKeeper();
+
+    if (is_readonly || !zookeeper)
+        throw Exception("Can't drop readonly replicated table (need to drop data in ZooKeeper as well)", ErrorCodes::TABLE_IS_READ_ONLY);
+
+    if (zookeeper->expired())
+        throw Exception("Table was not dropped because ZooKeeper session has expired.", ErrorCodes::TABLE_WAS_NOT_DROPPED);
+
+    if (is_drop_table)
+        replica_is_active_node = nullptr;
```

Comment:
Isn't it done in `StorageReplicatedMergeTree::shutdown()` which calls `ReplicatedMergeTreeRestartingThread::partialShutdown()`?